piecedirectory/piecedirectory_test.go

```diff
@@ -329,3 +335,61 @@ func testFlaggingPieces(ctx context.Context, t *testing.T, cl *client.Store) {
 	require.NoError(t, err)
 	require.Equal(t, 0, len(pcids))
 }
+
+func testReIndexMultiSector(ctx context.Context, t *testing.T, cl *client.Store) {
+	ctrl := gomock.NewController(t)
+	pr := mock_piecedirectory.NewMockPieceReader(ctrl)
+	pm := NewPieceDirectory(cl, pr, 1)
+	pm.Start(ctx)
+
+	// Create a random CAR file
+	carFilePath := CreateCarFile(t)
+	carFile, err := os.Open(carFilePath)
+	require.NoError(t, err)
+	defer carFile.Close()
+
+	carReader, err := car.OpenReader(carFilePath)
+	require.NoError(t, err)
+	defer carReader.Close()
+	carv1Reader, err := carReader.DataReader()
+	require.NoError(t, err)
+
+	pr.EXPECT().GetReader(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, fmt.Errorf("piece error")).Times(3)
+	pr.EXPECT().GetReader(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes().DoAndReturn(
+		func(_ context.Context, _ abi.SectorNumber, _ abi.PaddedPieceSize, _ abi.PaddedPieceSize) (pdTypes.SectionReader, error) {
+			_, err := carv1Reader.Seek(0, io.SeekStart)
+			return MockSectionReader{carv1Reader}, err
+		})
+
+	pieceCid := CalculateCommp(t, carv1Reader).PieceCID
+
+	// Add deal info for the piece - it doesn't matter what it is, the piece
+	// just needs to have at least one deal associated with it
+	d1 := model.DealInfo{
+		DealUuid:    uuid.New().String(),
+		ChainDealID: 1,
+		SectorID:    2,
+		PieceOffset: 0,
+		PieceLength: 0,
+	}
+
+	d2 := model.DealInfo{
+		DealUuid:    uuid.New().String(),
+		ChainDealID: 2,
+		SectorID:    3,
+		PieceOffset: 0,
+		PieceLength: 0,
+	}
+
+	err = cl.AddDealForPiece(ctx, pieceCid, d1)
+	require.NoError(t, err)
+
+	err = cl.AddDealForPiece(ctx, pieceCid, d2)
+	require.NoError(t, err)
+
+	err = pm.BuildIndexForPiece(ctx, pieceCid)
+	require.ErrorContains(t, err, "piece error")
+
+	err = pm.BuildIndexForPiece(ctx, pieceCid)
+	require.NoError(t, err)
```

Comment:
Let's add some comments here explaining why we expect an error / don't expect an error